CONFIG.md

```diff
@@ -0,0 +1,18 @@
+# ci-operator configuration file
```

Comment:
Can we use the upstream codegen to generate this file from the structs?

Reply:
@smarterclayton also mentioned this. I guess I will try what comes out of the tool, but I am skeptical how well can struct documentation serve as a reference on how to write JSON.

Reply:
Even godoc is probably good enough.

Reply:
Actually, if you add an examples section https://godoc.org/github.com/openshift/ci-operator/pkg/api is pretty much correct.

Reply:
I am not sure I would consider this very helpful if I wanted to write a config file... I mean yeah, the content is there if you know where to look, and it is correct, but the connection to how the config is written is *very* indirect if you are not familiar with how the thing works internally.